### PR TITLE
e2e: set version bundle version from defined bundles

### PR DESCRIPTION
**Pull request by fgimenez**
@kopiczko this is the version bundle version retrieval part, the rest in #714 is about parallel execution of WIP and current version bundle versions.

### CODE REVIEW

File: integration/framework.go [hunk 1]
```diff
@@ -457,3 +458,21 @@ func (f *framework) ApplyAWSConfigPatch(patch []PatchSpec, clusterName string) e
...
+
+func (f *framework) GetVersionBundleVersion(bundle []versionbundle.Bundle, vType string) string {
+	var output string
+	log.Printf("Tested version %q", vType)
+
+	for _, v := range bundle {
+		if (vType == "current" || vType == "") && !v.Deprecated && !v.WIP {
+			output = v.Version
+			break
+		}
```
Comment (kopiczko) [inline integration/framework.go]: This is possible that we have multiple non-deprecated non-WIP versions. This isn't deterministic. I think you can use data to take latest.

File: integration/framework.go [hunk 2]
```diff
@@ -457,3 +458,21 @@ func (f *framework) ApplyAWSConfigPatch(patch []PatchSpec, clusterName string) e
...
+		if vType == "wip" && v.WIP {
```
Comment (kopiczko) [inline integration/framework.go]: Similar situation to the above.

File: integration/framework.go [hunk 3]
```diff
@@ -457,3 +458,21 @@ func (f *framework) ApplyAWSConfigPatch(patch []PatchSpec, clusterName string) e
 
 	return nil
 }
+
+func (f *framework) GetVersionBundleVersion(bundle []versionbundle.Bundle, vType string) string {
```
Comment (kopiczko) [inline integration/framework.go]: This doesn't have to be a method of framework, but this is a minor thing.

But, wouldn't it be nice to validate vType? Or if you don't want to return error make it a typed constant and handle it in TestMain?

Comment (fgimenez) [reply integration/framework.go]: Sure, made function.

Ok, I've added validation and now the function returns an error too.

File: integration/framework.go (hunk 1 above)
Comment (fgimenez) [reply integration/framework.go]: Added code to return the latest non-deprecated non-wip for current.

File: integration/framework.go (hunk 2 above)
Comment (fgimenez) [reply integration/framework.go]: Same here, latest wip is returned.

**Comment by fgimenez** [comment]
@kopiczko comments addressed, PTAL

File: integration/framework.go [hunk 4]
```diff
@@ -457,3 +459,36 @@ func (f *framework) ApplyAWSConfigPatch(patch []PatchSpec, clusterName string) e
...
+		return "", fmt.Errorf("%q is not a valid version bundle version type", vType)
+	}
+
+	var output string
+	log.Printf("Tested version %q", vType)
+
+	// sort bundle by time to get the newest vbv.
+	s := versionbundle.SortBundlesByTime(bundle)
+	sort.Sort(s)
+	for i := len(s) - 1; i >= 0; i-- {
```
Comment (kopiczko) [inline integration/framework.go]: I think using reverse is a bit cleaner https://golang.org/pkg/sort/#Reverse

Comment (fgimenez) [reply integration/framework.go]: Yep much better :) done

**Pull request merged**